component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/totp/util/TOTPUtil.java

```diff
@@ -528,7 +535,9 @@ public static boolean isEnrolUserInAuthenticationFlowEnabled(AuthenticationConte
                 context.getProperty(TOTPAuthenticatorConstants.GET_PROPERTY_FROM_IDENTITY_CONFIG);
         //If the config file is not in registry and the it is super tenant, getting the property from local.
         // Else getting it from context.
-        if ((getPropertiesFromIdentityConfig != null ||
+        if (!TOTPAdaptiveScriptEnabled) {
+            return false;
+        } else if ((getPropertiesFromIdentityConfig != null ||
```

Comment:
if TOTP is enabled in the script. No need to check in the xml config. Just allow it.